Restore and refactor facts filtering

melck opened the pull request:
that has been removed in #708

Fixes #714

gdubicki commented:
Cool, it’s good to get rid of a technology if it’s used for one thing only. :) 

So it will be functionally the same as before?

gdubicki reviewed (changes requested):
Can you please also remove coffeescript from dev.py and possibly other references?

melck commented:
Hello @gdubicki ,

I got rid of all related coffeescript things. Since the file `dev.py` was only useful to handle coffeescript compilation, i removed it to use default flask command. I updated dev docs in consequences.

Regarding functionnality, it will be the same but i have added a sightly delay (250ms) for filtering to avoid weird freeze of browsers when typing to fast.
